### PR TITLE
Mtl: Add base code to bring up interfaces

### PR DESCRIPTION
**Pull request by softwarecki**
Added clock definitions, dma channels configuration and updated configuration options of the mtl board.

### CODE REVIEW

File: src/platform/intel/ace/lib/dma.c
Comment (abonislawski) [inline src/platform/intel/ace/lib/dma.c:19]: why we need this?

File: src/platform/intel/ace/lib/dma.c
Comment (abonislawski) [inline src/platform/intel/ace/lib/dma.c:21]: all channels data should be taken from DT

File: src/platform/intel/ace/lib/dma.c
Comment (abonislawski) [inline src/platform/intel/ace/lib/dma.c:99]: Not z_dev but the whole GPDMA0 depends on DT
same apply to all DMAs

File: src/platform/intel/ace/lib/clk.c
Comment (abonislawski) [inline src/platform/intel/ace/lib/clk.c:17]: why we need this file?

Comment (softwarecki) [reply src/platform/intel/ace/lib/clk.c:17]: It is used by clock_get_freq function. For example called by basefw_config function from base_fw module. Zepphyr wrapper also using it in platform_dai_timestamp function.

File: src/platform/intel/ace/lib/dma.c
Comment (softwarecki) [reply src/platform/intel/ace/lib/dma.c:26]: The same configuration option is used in the zephyrs dma driver. Let's keep this option in case of a configuration change in the future.

**Review by lgirdwood — APPROVED** [review]
LGTM, but I wont merge until @abonislawski and @softwarecki are aligned.

**Comment by softwarecki** [comment]
This PR requires memory windows 1 setup in the Zephyr: https://github.com/zephyrproject-rtos/zephyr/pull/50919

**Comment by lgirdwood** [comment]
> 

ok, pls ping the PR when teh Zephyr change is merged.

File: app/boards/intel_adsp_ace15_mtpm.conf
Comment (abonislawski) [inline app/boards/intel_adsp_ace15_mtpm.conf:41]: HDA backend is not supported in MTL, please remove

File: app/boards/intel_adsp_ace15_mtpm.conf
Comment (abonislawski) [inline app/boards/intel_adsp_ace15_mtpm.conf:40]: CONFIG_LOG_BACKEND_ADSP is not needed too, we can leave only mtrace enabled by default

**Review by abonislawski — CHANGES_REQUESTED** [review]

File: src/platform/intel/ace/lib/dma.c
Comment (marcinszkudlinski) [inline src/platform/intel/ace/lib/dma.c:27]: magic number

File: src/platform/intel/ace/lib/dma.c
Comment (marcinszkudlinski) [inline src/platform/intel/ace/lib/dma.c:54]: more magic numbers

**Review by marcinszkudlinski — CHANGES_REQUESTED** [review]

**Review by kv2019i — CHANGES_REQUESTED** [review]
A few minor things, but looks good otherwise.

File: src/platform/intel/ace/lib/clk.c
Comment (lyakh) [inline src/platform/intel/ace/lib/clk.c:22]: `platform_clocks_info` isn't cached

File: src/platform/intel/ace/lib/dma.c
Comment (lyakh) [inline src/platform/intel/ace/lib/dma.c:156]: I don't think you need the `struct dma *` cast - it's already the correct type. Besides since this is Zephyr-only, you actually don't need `cache_to_uncache_init()` either

**Comment by softwarecki** [comment]
This PR requires following PR merged in Zephyr:
https://github.com/zephyrproject-rtos/zephyr/pull/49458
https://github.com/zephyrproject-rtos/zephyr/pull/48949

File: src/platform/intel/ace/lib/dma.c
Comment (marcinszkudlinski) [reply src/platform/intel/ace/lib/dma.c:27]: DW_DMA_BUFFER_ALIGNMENT

**Comment by lgirdwood** [comment]
> This PR requires following PR merged in Zephyr: [zephyrproject-rtos/zephyr#49458](https://github.com/zephyrproject-rtos/zephyr/pull/49458) [zephyrproject-rtos/zephyr#48949](https://github.com/zephyrproject-rtos/zephyr/pull/48949)

48949 is now merged, in the meantime can we have an update with the review comments addressed. This will get us ready for the final Zephyr patch being merged. Thanks !

**Review by lyakh — APPROVED** [review]

**Review by marcinszkudlinski — APPROVED** [review]

**Comment by kv2019i** [comment]
@softwarecki @wszypelt License-server issue hit in the CI run.

**Comment by marc-hb** [comment]
> @softwarecki @wszypelt License-server issue hit in the CI run.

QB or Jenkins? Always copy the URL, it's hard/impossible to find after a force push.

**Comment by abonislawski** [comment]
One required PR left: #6396 

Please wait for confirmation that this PR is working with validation tests before merging.

**Comment by softwarecki** [comment]
I'm currently investigating the problem with running the pipeline. so please hold on with merging this PR

**Review by abonislawski — APPROVED** [review]
Good to merge!

**Comment by kv2019i** [comment]
Proceeding with merge. This uncovers some compiler warnings when built with Zephyr SDK, but this these were in existing code, so proceeding with merge. @softwarecki @abonislawski can you look at https://github.com/thesofproject/sof/actions/runs/3272915314/jobs/5384495135

**Comment by kv2019i** [comment]
> This uncovers some compiler warnings when built with Zephyr SDK, but this these were in existing code, so proceeding with merge. @softwarecki @abonislawski can you look at https://github.com/thesofproject/sof/actions/runs/3272915314/jobs/5384495135

-> https://github.com/thesofproject/sof/pull/6439

**Pull request merged**
